File: LeetCode/Explore/TopInterview/Amazon/SortingSearching/167_two_sum_II_input_array_is_sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        mp = dict()
        vp = list()
        
        for i, num in enumerate(numbers):
          if target - num in mp:
            vp.append(mp[target - num])
            vp.append(i + 1)
          mp[num] = i + 1
          
        return vp

File: LeetCode/Explore/TopInterview/Amazon/SortingSearching/167_two_sum_II_input_array_is_sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
      unordered_map<int, int> mp;
      vector<int> vp;
      for (int i = 0; i < numbers.size(); i++) {
        if (mp.count(target - numbers[i])) {
          vp.push_back(mp[target - numbers[i]]);
          vp.push_back(i + 1);
        }
        mp[numbers[i]] = i + 1;
      }
      return vp;
    }
};
